src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -55,44 +57,78 @@ static void Main(string[] args)
+        /// <param name="level">Specifies the level of granuality to use for the output: either namepspace, class or function</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
+        private static void GetTests(IList<string> dlls, Level level)
+        {
+            foreach(string dll in dlls)
+            {
+                Assembly assembly = null;
+                try
                 {
-                    writer = new StreamWriter(Console.OpenStandardOutput());
-                }
-                else
+                    assembly = Assembly.LoadFrom(dll);
+                }catch(FileNotFoundException e)
                 {
-                    writer = new StreamWriter(outputDirectory + name +".txt", true);
+                    Console.Error.WriteLine("{0}: file {1} was not found on system", e.GetType(), dll);
+                    continue;
                 }
-                foreach (string dll in dllDirectories)
+               
+                foreach (Type type in assembly.GetTypes())
                 {
-                    if (File.Exists(dll))
-                    {
-                        writer.Write("{0}|{1}", dll, level + Environment.NewLine);
-                    }
-                    else
+                    foreach (MethodInfo methodInfo in type.GetMethods())
                     {
-                        Console.Error.WriteLine("Error: Unable to append {0} as it cannot be found on the system", dll);
+                        var attributes = methodInfo.GetCustomAttributes(true);
+                        foreach (var attr in attributes)
+                        {
+                            if (attr is NUnit.Framework.TestAttribute)
+                            {                               
+                                string value = "";
+                                
+                                switch (level)
+                                {
+                                    case Level.Namespace:
+                                        value = type.Namespace;
+                                        break;
+                                    case Level.Class:
+                                        value = methodInfo.DeclaringType.ToString();
+                                        break;
+                                    case Level.Function:
+                                        string method = methodInfo.ToString().Substring(methodInfo.ToString().IndexOf(" ", StringComparison.OrdinalIgnoreCase) + 1);
+                                        value = methodInfo.DeclaringType + "." + method;
+                                        break;
+                                }
+                                StreamWriter writer;
+                                if (_options.Output== null)
```

Comment:
Also, to fix the 2 code analysis warnings about calling Dispose(), when you move these lines into main(), you can do this in main() add a finally block to the try (and declare the writer before the try) and in the finally block do this:
```
finally
{
    if (writer != null)
    {
        writer.Dispose();
    }
}
```